src/main/java/com/gooddata/http/client/LoginSSTRetrievalStrategy.java

```diff
@@ -43,32 +38,43 @@
 
     private final HttpHost httpHost;
 
-    private final HttpClient httpClient;
-
     /**
      * Construct object.
+     * @deprecated Use {@link #LoginSSTRetrievalStrategy(String, String)}}
      * @param httpClient HTTP client
      * @param httpHost http host
      * @param login user login
      * @param password user password
      */
+    @Deprecated
     public LoginSSTRetrievalStrategy(final HttpClient httpClient, final HttpHost httpHost, final String login, final String password) {
-        notNull(httpClient, "HTTP Client cannot be null");
-        notNull(httpHost, "HTTP host cannot be null");
         notNull(login, "Login cannot be null");
         notNull(password, "Password cannot be null");
         this.login = login;
         this.password = password;
         this.httpHost = httpHost;
-        this.httpClient = httpClient;
+    }
+
+    /**
+     * Construct object.
+     * @param login user login
+     * @param password user password
+     */
+    @SuppressWarnings("deprecated")
+    public LoginSSTRetrievalStrategy(final String login, final String password) {
+        this(null, null, login, password);
+    }
+
+    HttpHost getHttpHost() {
+        return httpHost;
     }
 
     @Override
-    public String obtainSst() throws IOException {
+    public String obtainSst(final HttpClient httpClient, final HttpHost httpHost) throws IOException {
         log.debug("Obtaining STT");
         final HttpPost postLogin = new HttpPost(LOGIN_URL);
         try {
-            final HttpEntity requestEntity = new StringEntity(createLoginJson(login, password), ContentType.APPLICATION_JSON);
+            final HttpEntity requestEntity = new StringEntity(createLoginJson(), ContentType.APPLICATION_JSON);
             postLogin.setEntity(requestEntity);
             postLogin.setHeader("Accept", ContentType.APPLICATION_JSON.toString());
```

Comment:
please consider to use [HttpHeaders constant](https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpHeaders.html)